loopy/schedule/tools.py

```diff
@@ -381,6 +382,37 @@ def _check_for_access_races(map_a, insn_a, map_b, insn_b, knl, callables_table):
             if dt == isl.dim_type.in_:
                 tag, = filter_iname_tags_by_type(knl.inames[name].tags,
                                                  HardwareConcurrentTag)
+                if any(not aff.plain_is_zero()
+                       for _, aff in (knl
+                                      .get_iname_bounds(name)
+                                      .lower_bound_pw_aff
+                                      .get_pieces())
+                       ):
+                    # Non-zero hardware inames are offseted in loopy's codegen. See
+                    # https://github.com/inducer/loopy/issues/600#issuecomment-1104066735
+                    # => Accordingly offset the access maps
+                    lbound_val = (static_min_of_pw_aff(knl
+                                                       .get_iname_bounds(name)
+                                                       .lower_bound_pw_aff,
+                                                       constants_only=True)
+                                  .get_constant_val().to_python())
```

Comment:
This now has a two-sources-of-truth problem. What if it's inconsistent with codegen? (Which, in fact, it already is?) I think it would be better to record the applied base in `codegen_state` and use that here.

Reply:
Factor the logic into a function?

Reply:
Yep!

Reply:
Done in [24d05e7](https://github.com/inducer/loopy/pull/816/commits/24d05e7acd86de6537de782e0802a3e211db2493).

(CI should pass once #821 is merged.)